### PR TITLE
[APB-3618][PT]: Integrate with Email and send when user has created Agent-Services-Account

### PR DESCRIPTION
**Pull request by JTY009**

### CODE REVIEW

File: app/uk/gov/hmrc/agentsubscription/connectors/EmailConnector.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  override val kenshooRegistry: MetricRegistry = metrics.defaultRegistry
+
+  def sendEmail(emailInformation: EmailInformation)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Unit] =
+    monitor(s"Send-Email-${emailInformation.templateId}") {
+      http
+        .POST[EmailInformation, HttpResponse](new URL(s"$baseUrl/hmrc/email").toString, emailInformation)
+        .map(_ => ())
+    }.recover {
+      case e => throw new Exception(s"sending email failed: $e")
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentsubscription/connectors/EmailConnector.scala]: is it correct to throw exception here ? 
I guess by now the account has been created but due to this exception the frontend receives an error saying that there is some tech error which confuses users

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentsubscription/connectors/EmailConnector.scala]: i think we can log an error but ignore the exception ?

Comment (JTY009) [reply app/uk/gov/hmrc/agentsubscription/connectors/EmailConnector.scala]: But the user expects an email sent to them when they see the final page

File: app/uk/gov/hmrc/agentsubscription/connectors/EmailConnector.scala [hunk 2]
```diff
@@ -45,6 +45,6 @@ class EmailConnectorImpl @Inject() (@Named("email-baseUrl") baseUrl: URL, http:
         .POST[EmailInformation, HttpResponse](new URL(s"$baseUrl/hmrc/email").toString, emailInformation)
         .map(_ => ())
     }.recover {
-      case e => throw new Exception(s"sending email failed: $e")
+      case e => Logger.error(s"sending email failed: $e")
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentsubscription/connectors/EmailConnector.scala]: sorry, should this be at WARN level ?

**Pull request merged**